### PR TITLE
link detection fix

### PR DESCRIPTION
**Pull request by Dmole**
link detection fix

### CODE REVIEW

**Comment by xoriole** [comment]
ok to test

File: src/tribler.sh [hunk 1]
```diff
@@ -2,8 +2,8 @@
 # Run Tribler from source tree
 
 script_path() {
-  DIR="${1%/*}"
-  (cd "$DIR" && echo "$(pwd -P)")
+  FILE="$(readlink -f "$1")"
```
Comment (xoriole) [inline src/tribler.sh:5]: While using `readlink -f` makes it possible to run tribler.sh from any path, the command is not available on macOS which is why we avoided using it earlier.

**Review by xoriole — CHANGES_REQUESTED** [review]
Thank you for the PR. It works nicely on Linux but not on macOS so can not accept as it is, therefore requesting changes

File: src/tribler.sh (hunk 1 above)
Comment (Dmole) [reply src/tribler.sh:5]: readlink must have been added for me by brew, I'll make it optional in the PR.

**Comment by drew2a** [comment]
I've converted this PR to "draft" as it is still in WIP state.

**Comment by Dmole** [comment]
> I've converted this PR to "draft" as it is still in WIP state.

Let me know what needs to change to get it merged.

**Comment by drew2a** [comment]
> Let me know what needs to change to get it merged.

As far as I know -- it is necessary to address all @xoriole requests and re-request his review 👍
image

**Comment by drew2a** [comment]
@Dmole please, update your branch to the latest main. The current test failures are caused by changes from https://github.com/Tribler/tribler/pull/6799.

**Comment by drew2a** [comment]
Give me 1 hour and I'll fix the test run (no action needed from your side).

**Comment by drew2a** [comment]
Ok, I've fixed test runs. Now tests from the latest main should work fine.

**Comment by xoriole** [comment]
@Dmole I have tested the PR on both Linux and macOS, and it works. Please rebase the PR to get it merged.

**Review by xoriole — APPROVED** [review]

**Pull request merged**